src/main/java/groovy/lang/MetaClassImpl.java

```diff
@@ -2207,11 +2210,34 @@ private static MetaMethod getCategoryMethodSetter(Class sender, String name, boo
 //                    element = new MetaBeanProperty(mp.getName(), mp.getType(), null, mp.getSetter());
 //                }
             }
+
+            if (!ALLOW_ILLEGAL_ACCESS_PROPERTIES) {
+                if (element instanceof MetaBeanProperty) {
+                    MetaBeanProperty mbp = (MetaBeanProperty) element;
+                    boolean getterAccessible = canAccessLegally(mbp.getGetter());
+                    boolean setterAccessible = canAccessLegally(mbp.getSetter());
+
+                    if (!(getterAccessible && setterAccessible)) continue;
+                }
+            }
+
             ret.add(element);
         }
         return ret;
     }
 
+    private static final boolean ALLOW_ILLEGAL_ACCESS_PROPERTIES = SystemUtil.getBooleanSafe("groovy.allow.illegal.access.properties");
```

Comment:
This all looks good - my only suspicion is that some folks will want to turn this feature on very selectively.
We could make an instance property which is set by the system property but could be toggled on an individual metaclass basis. I am also wondering whether "groovy.permissive.property.access" would be a better name. In future JDK versions the access might not be allowed at all - for us it is indicating that we will attempt to access the property.

Reply:
Thanks for you reviewing the PR, Paul :)

I wish I had caught the points:
1) Rename `groovy.allow.illegal.access.properties`  to `groovy.permissive.property.access`
2) Add an instance property whose value is initialized by the system property `groovy.permissive.property.access`, we use the instance property to control the illegal access to properties